### ISSUE TITLE
Incorrect name mangling of nested record fields?

### ISSUE THREAD

**Issue by drathier** [purerl/purerl#30]
EDIT: read bottom up

purs version: 0.15.4
purerl version: 0.0.18
os: mac

where-bound variable gets renamed to `Op@prime@13`
but usages ask for `opStart=>('op\'1'()), opLetter=>('op\'1'())`

.spago/parsing/v6.0.2-erl1/src/Text/Parsing/Parser/Language.purs
```haskell
haskellStyle :: LanguageDef
haskellStyle = LanguageDef (unGenLanguageDef emptyDef)
                { commentStart    = "{-"
                , commentEnd      = "-}"
                , commentLine     = "--"
                , nestedComments  = true
                , identStart      = letter
                , identLetter     = alphaNum <|> oneOf ['_', '\'']
                , opStart         = op'
                , opLetter        = op'
                , reservedOpNames = []
                , reservedNames   = []
                , caseSensitive   = true
                }
  where
    op' :: forall m . (Monad m) => ParserT String m Char
    op' = oneOf [':', '!', '#', '$', '%', '&', '*', '+', '.', '/', '<', '=', '>', '?', '@', '\\', '^', '|', '-', '~']
```
```haskell
emptyDef :: LanguageDef
emptyDef = LanguageDef
            { commentStart:    ""
            , commentEnd:      ""
            , commentLine:     ""
            , nestedComments:  true
            , identStart:      letter <|> char '_'
            , identLetter:     alphaNum <|> oneOf ['_', '\'']
            , opStart:         op'
            , opLetter:        op'
            , reservedOpNames: []
            , reservedNames:   []
            , caseSensitive:   true
            }
  where
    op' :: forall m . (Monad m) => ParserT String m Char
    op' = oneOf [':', '!', '#', '$', '%', '&', '*', '+', '.', '/', '<', '=', '>', '?', '@', '\\', '^', '|', '-', '~']
```

**Comment by drathier** [comment]
Found another one which doesn't include prime, and which correctly mangles the name in some places:
`MaybeGetA2@50 = (MaybeGetA@48( ....` and usage site `#{given=>(maybeGetD1(#{d=>true})), expected=>{ nothing }}`

```erlang
tests() -> 
  MaybeGetD@45 = fun MaybeGetD(DictGetFieldIfItExists) ->
    GetFieldIfItExists6@51 = (?MEMOIZE((getFieldIfItExists(DictGetFieldIfItExists)))),
    fun (R) ->
      ((GetFieldIfItExists6@51({ proxy }))(R))
    end
  end,
  MaybeGetD1@46 = (MaybeGetD@45((?MEMOIZE((getFieldIfItExists4((dIsSymbol()))))))),
  MaybeGetD2@47 = (MaybeGetD@45((?MEMOIZE((getFieldIfItExists2((?MEMOIZE((containsFieldRLCons2((?MEMOIZE((containsFieldRLCons2((containsFieldRLCons1()))))))))), (dIsSymbol()))))))),
  MaybeGetA@48 = fun MaybeGetA(DictGetFieldIfItExists) ->
    GetFieldIfItExists6@52 = (?MEMOIZE((getFieldIfItExists(DictGetFieldIfItExists)))),
    fun (R) ->
      ((GetFieldIfItExists6@52({ proxy }))(R))
    end
  end,
  MaybeGetA1@49 = (MaybeGetA@48((?MEMOIZE((getFieldIfItExists2((?MEMOIZE((containsFieldRLCons((aIsSymbol()))))), (aIsSymbol()))))))),
  MaybeGetA2@50 = (MaybeGetA@48((?MEMOIZE((getFieldIfItExists4((aIsSymbol()))))))),
  (etTest_test@ps:describe(<<"Lesslie.BlackMagic"/utf8>>, (array:from_list([(etTest_test@ps:testTable(<<"GetFieldIfItExists 'a'"/utf8>>, (array:from_list([#{given=>(maybeGetA1(#{a=>1})), expected=>{ just, 1 }}, #{given=>((MaybeGetA@48((?MEMOIZE((getFieldIfItExists3((aIsSymbol())))))))(#{})), expected=>{ nothing }}, #{given=>(maybeGetA2(#{b=>1})), expected=>{ nothing }}, #{given=>(maybeGetA2(#{a=>true})), expected=>{ nothing }}, #{given=>((MaybeGetA@48((?MEMOIZE((getFieldIfItExists5((aIsSymbol())))))))(#{a=>true, b=>1, c=>1})), expected=>{ nothing }}, #{given=>(maybeGetA1(#{a=>1, b=>1, c=>1})), expected=>{ just, 1 }}, #{given=>(maybeGetA1(#{b=>1, c=>1, a=>1})), expected=>{ just, 1 }}])), fun (V) ->
    case V of
      (#{given:=_@38, expected:=_@39}) -> (equal(_@39, _@38))
    end
  end)), (etTest_test@ps:testTable(<<"GetFieldIfItExists 'd'"/utf8>>, (array:from_list([#{given=>((MaybeGetD@45((?MEMOIZE((getFieldIfItExists2((containsFieldRLCons1()), (dIsSymbol())))))))(#{d=>1})), expected=>{ just, 1 }}, #{given=>((MaybeGetD@45((?MEMOIZE((getFieldIfItExists3((dIsSymbol())))))))(#{})), expected=>{ nothing }}, #{given=>(maybeGetD1(#{b=>1})), expected=>{ nothing }}, #{given=>(maybeGetD1(#{d=>true})), expected=>{ nothing }}, #{given=>((MaybeGetD@45((?MEMOIZE((getFieldIfItExists5((dIsSymbol())))))))(#{d=>true, b=>1, c=>1})), expected=>{ nothing }}, #{given=>(maybeGetD2(#{d=>1, b=>1, c=>1})), expected=>{ just, 1 }}, #{given=>(maybeGetD2(#{b=>1, c=>1, d=>1})), expected=>{ just, 1 }}])), fun (V) ->
    case V of
      (#{given:=_@40, expected:=_@41}) -> (equal(_@41, _@40))
    end
  end))]))))
.
```

**Comment by drathier** [comment]
[try purescript link](https://try.purescript.org/?code=LYewJgrgNgpgBAWQIYEsB2cDuALGAnGAKEJWAAcQ8AXOAESSqQDpkBPAIyNIuroeYDKrYOxBQS5SjQAKeUkwBKITABkUAZyoSeNBTADGlMEwCqadUgBmXSbwAqrMjCayQAD1bF9UJOvVwAYRA0RnR1ADEUGCgwBRU4AC4EuEtKJCgoOABrJmy4AFoAPjgHJwLipVUNGlL4IsDgzTxUEMJvX38gkJaIqJi4lL6wOBpMnHwiODhDbrDI6Nj4pLhXD0GF8pW8d1YRzdXdzPqAIRAxGCQ0YjDGNH1Jhtnzef74yyG9gDkUMdwCQimM1CzyGAwA%2BnAIRCALwpdLqIjRBFwG6Xe4AuAASXUQhEYnWMTg0MKGK6wN6CwG7w2NAAFF1-NTCTRQlAAJRYP4PIE9F6LSECmEjPAQRFQZGou4PWmPcl8qkfFmocRTDnE0nBOWgt6KuD0xqQpnDMFKn4c8b-QGa3nauBkPZCnlzW32ukHRLJd2stleHx%2BOAAcRgVD5mMsmKoAFE3NV-MtUs0MtlcllNrVNpUSo46sUGVRmugtO1-UGQ0MwxHo7GCcMaHhORMMQBzYOh8NRmOaOOe7ZrI0ZgxGOD1%2BpsTgja7mW73PWVOwgNSaYdwKAAGllNspOppK-X2NxokyRrVxVLbcrnaojN19YtDxbZYWFY71f7BEMeGG0IxUxQlmm1rOluNZ6u6yxehytJgT2OwrhyVC4FcUzIXAABSEBLrSEDmFYMClnqBCWLA%2BhUAe%2BLHsOg6fj6KFIg8KGfCACHoE2xBUDAXZEhisBaChwBIBweHBrQHopGkSZ4LkZ7lu2VacQARGA8lYiEy7EnAegfsMI7FGO8CYq0fECZwpYifWsIPueL6cVBva7OBdlwIp8kcngGLoBiADacD8YJplwAA3nA7DJAAjHAAC%2BGLrr5JnCYFcBgMk%2BaipFGIALqEEAA)
```purescript
module Main where

import Data.Maybe
import Data.Symbol
import Prim.RowList
import Record.Unsafe
import Type.Proxy

class ContainsFieldRL :: forall k. k -> Type -> RowList Type -> Constraint
class ContainsFieldRL field t l where
  containsFieldRL :: Proxy field -> Proxy t -> Proxy l -> Boolean

instance
  ContainsFieldRL field t Nil where
  containsFieldRL _ _ _ = false
else instance
  IsSymbol field =>
  ContainsFieldRL field t (Cons field t tail) where
  containsFieldRL _ _ _ = true
else instance
  ( ContainsFieldRL field t tail
  ) =>
  ContainsFieldRL field t (Cons _field _t tail) where
  containsFieldRL p t _ = containsFieldRL p t (Proxy :: Proxy tail)

class GetFieldIfItExists :: forall k. k -> Type -> Row Type -> Constraint
class GetFieldIfItExists field t r where
  getFieldIfItExists :: Proxy field -> Record r -> Maybe t

instance (RowToList r l, ContainsFieldRL field t l, IsSymbol field) => GetFieldIfItExists field t r where
  getFieldIfItExists field record =
    if containsFieldRL field (Proxy :: Proxy t) (Proxy :: Proxy l) then
      Just (unsafeGet (reflectSymbol field) record)
    else
      Nothing

tests =
  let
    maybeGetD :: forall r. GetFieldIfItExists "d" Int r => Record r -> Maybe Int
    maybeGetD r = getFieldIfItExists (Proxy :: Proxy "d") r
  in
  [ maybeGetD { b: 1 }
  , maybeGetD { d: true }
  ]
```

**Comment by drathier** [comment]
```erlang
function maybeGetD1/1 undefined
%  116|   (array:from_list([(maybeGetD1(#{b=>1})), (maybeGetD1(#{d=>true}))]))
%     |                                             ^
```

```erlang
-spec getFieldIfItExists1(any(),any(),any()) -> any().
getFieldIfItExists1(_@23,_@24,_@25) -> ((((getFieldIfItExists1())(_@23))(_@24))(_@25)).
-spec tests() -> array:array(data_maybe_Maybe(integer())).
tests() -> 
  MaybeGetD@29 = fun MaybeGetD(DictGetFieldIfItExists) ->
    GetFieldIfItExists2@31 = (?MEMOIZE((getFieldIfItExists(DictGetFieldIfItExists)))),
    fun (R) ->
      ((GetFieldIfItExists2@31({ proxy }))(R))
    end
  end,
  MaybeGetD1@30 = (MaybeGetD@29((?MEMOIZE((getFieldIfItExists1(undefined, (?MEMOIZE((containsFieldRLCons1((containsFieldRLNil()))))), #{reflectSymbol=>fun (_) ->
    <<"d"/utf8>>
  end})))))),
  (array:from_list([(maybeGetD1(#{b=>1})), (maybeGetD1(#{d=>true}))])) %% error is here, maybeGetD1
.
```

**Comment by nwolverson** [comment]
I believe this is fixed in latest `purs` (0.15.6) - my example seems to build fine again, though possibly I'm tricking myself. Can you check?

**Comment by nwolverson** [comment]
Closing this as it should be fixed.
